contracts/stake/src/contract.rs

```diff
@@ -152,4 +191,32 @@ impl StakeContract {
 
         bytes
     }
+
+    pub fn add_owner(&mut self, owner: PublicKey) -> Result<(), Error> {
+        if self.is_owner(&owner)? {
+            return Ok(());
+        }
+        self.owners.insert(owner, true)?;
+        Ok(())
+    }
+
+    pub fn is_owner(&self, owner: &PublicKey) -> Result<bool, Error> {
+        Ok(self.owners.get(owner)?.is_some())
+    }
+
+    pub fn check_ownership(&self, owner: &PublicKey) -> Result<bool, Error> {
+        Ok(self.whitelist.get(owner)?.is_some())
+    }
+
+    pub fn insert_whitelist(&mut self, staker: PublicKey) -> Result<(), Error> {
+        if self.is_whitelisted(&staker)? {
+            return Ok(());
+        }
+        self.whitelist.insert(staker, true)?;
+        Ok(())
+    }
+
+    pub fn is_whitelisted(&self, staker: &PublicKey) -> Result<bool, Error> {
```

Comment:
Wait... This is the same as `check_ownership`, just with a different argument name.